lib/rubygems/vendor/securerandom/lib/random/formatter.rb

```diff
@@ -0,0 +1,373 @@
+# -*- coding: us-ascii -*-
+# frozen_string_literal: true
+
+# == \Random number formatter.
+#
+# Formats generated random numbers in many manners. When <tt>'random/formatter'</tt>
+# is required, several methods are added to empty core module <tt>Random::Formatter</tt>,
+# making them available as Random's instance and module methods.
+#
+# Standard library Gem::SecureRandom is also extended with the module, and the methods
+# described below are available as a module methods in it.
+#
+# === Examples
+#
+# Generate random hexadecimal strings:
+#
+#   require 'random/formatter'
+#
+#   prng = Random.new
+#   prng.hex(10) #=> "52750b30ffbc7de3b362"
+#   prng.hex(10) #=> "92b15d6c8dc4beb5f559"
+#   prng.hex(13) #=> "39b290146bea6ce975c37cfc23"
+#   # or just
+#   Random.hex #=> "1aed0c631e41be7f77365415541052ee"
+#
+# Generate random base64 strings:
+#
+#   prng.base64(10) #=> "EcmTPZwWRAozdA=="
+#   prng.base64(10) #=> "KO1nIU+p9DKxGg=="
+#   prng.base64(12) #=> "7kJSM/MzBJI+75j8"
+#   Random.base64(4) #=> "bsQ3fQ=="
+#
+# Generate random binary strings:
+#
+#   prng.random_bytes(10) #=> "\016\t{\370g\310pbr\301"
+#   prng.random_bytes(10) #=> "\323U\030TO\234\357\020\a\337"
+#   Random.random_bytes(6) #=> "\xA1\xE6Lr\xC43"
+#
+# Generate alphanumeric strings:
+#
+#   prng.alphanumeric(10) #=> "S8baxMJnPl"
+#   prng.alphanumeric(10) #=> "aOxAg8BAJe"
+#   Random.alphanumeric #=> "TmP9OsJHJLtaZYhP"
+#
+# Generate UUIDs:
+#
+#   prng.uuid #=> "2d931510-d99f-494a-8c67-87feb05e1594"
+#   prng.uuid #=> "bad85eb9-0713-4da7-8d36-07a8e4b00eab"
+#   Random.uuid #=> "f14e0271-de96-45cc-8911-8910292a42cd"
+#
+# All methods are available in the standard library Gem::SecureRandom, too:
+#
+#   Gem::SecureRandom.hex #=> "05b45376a30c67238eb93b16499e50cf"
+
+module Random::Formatter
```

Comment:
Just noticed this wasn't namespaced. I think that will cause redefinition warnings.

Reply:
Good catch! I think we'll need to tweak automatiek to handle this.

Reply:
Yeah, secure random is weird. Upstream is aware but unclear what they'll do: https://github.com/ruby/securerandom/pull/29